clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/converters/DeployAppengineAtomicOperationConverter.groovy

```diff
@@ -22,16 +22,43 @@ import com.netflix.spinnaker.clouddriver.appengine.deploy.ops.DeployAppengineAto
 import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation
 import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperations
 import com.netflix.spinnaker.clouddriver.security.AbstractAtomicOperationsCredentialsSupport
+import com.netflix.spinnaker.clouddriver.appengine.deploy.exception.AppengineDescriptionConversionException
 import org.springframework.stereotype.Component
+import org.springframework.beans.factory.annotation.Autowired
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
 
 @AppengineOperation(AtomicOperations.CREATE_SERVER_GROUP)
 @Component
 class DeployAppengineAtomicOperationConverter extends AbstractAtomicOperationsCredentialsSupport {
+  @Autowired
+  ObjectMapper objectMapper = new ObjectMapper();
+
   AtomicOperation convertOperation(Map input) {
     new DeployAppengineAtomicOperation(convertDescription(input))
   }
 
   DeployAppengineDescription convertDescription(Map input) {
-    AppengineAtomicOperationConverterHelper.convertDescription(input, this, DeployAppengineDescription)
+    DeployAppengineDescription description = AppengineAtomicOperationConverterHelper.convertDescription(input, this, DeployAppengineDescription)
+
+    if (input.artifact) {
+      description.artifact = objectMapper.copy().convertValue(input.artifact, Artifact)
+      switch (description.artifact.type) {
```

Comment:
Can we deploy from a git artifact or is that not functional yet?

Reply:
This seems like you can't have a gcs artifact as well as a docker one in the same deploy? I think there should be an artifact representing the app.yml, and another representing the docker image. The former can be from gcs/git/s3/etc...

Reply:
True.  In the current implementation the `app.yaml` is expected to be entered in a free-text field in Deck and the content POSTed as part of the request body which is written to a temp app.yaml by DeployAppengineAtomicOperation.

Since this PR's purpose is to simply move logic from orca to clouddriver I propose tackling the handling of multiple artifacts in a new patchset.  Do you prefer I put it all in this commit or are you OK with a separate PR for that functionality?

Reply:
That's fine with me.